### DISCUSSION TITLE
is there a way to get dbnames of a column?

### DISCUSSION THREAD

**Discussion by volkerrichert** [prisma/prisma#22010] [Q&A] [answered]
### Question

I'm using prisma in a larger Project. Normaly I'm using the normal query but in some cases i have to use a raw query.

Is there a way to get the column names of a property which is mapped to another name like `  id  String @id @default(cuid()) @map("cid")`?

When i build my raw query i'd like to use something like "prima.model.fields.id.dbName" to map model name to db names. Esp. during refactoring with changed column names it could reduce errors.

**Comment by ludralph** [comment, marked as answer]
Hi @volkerrichert 👋 

Thank you for raising this question.

Given a model like below:
```prisma
model User {
  id Int @id @default(autoincrement()) @map("user_id")
  name String @map("full_name")
  email String @unique @map("email_address")
  createdAt DateTime @default(now()) @map("created_at")
  updatedAt DateTime @updatedAt @map("updated_at")
}
```
You can get the database names of the column by using Prisma's internal `DMMF` API.

```js
const res = await Prisma.dmmf.datamodel.models.flatMap(model => model.fields.map(a=>a.dbName))
```
Notice the `dbName` field in the code above. This will output a result of the form:
```
[ 'user_id', 'full_name', 'email_address', 'created_at', 'updated_at' ]
```
Notice how the result is an array of the mapped column names. **You should however note that the DMMF API is internal to Prisma and could have breaking changes in any release without warning.**

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by volkerrichert** [reply]
ok, THX. Maybe it could be added to Prisma.getExtensionContext.
